src/quest/historian.py

```diff
@@ -659,6 +668,8 @@ async def handle_internal_event(self, name, identity, action, *args, **kwargs):
                 kwargs=kwargs,
                 result=result
             ))
+            if action != 'get':
+                await self._handle_resource_update(identity)
```

Comment:
This feels too cludgy to me. I think we need a design review on the idea.

- Are there other resources that would have a similar need? Is it only the state object?
- Is there another way to present state data? What if state is readonly outside the workflow - does this change the way we would implement this?

Specific-case branches in the code are often an indication the design is bad. So I would want a well-reasoned, solid motivation for why this use-case needs to exist and be implemented this way.

Reply:
I agree that this could warrant a deeper investigation into how this problem could be handled. However I want to note that this condition in line 671 acts on when .get() is called on any resource, not just when state.get() is called. It seems to me that this makes sense- that the user can read the value of a resource anytime without triggering a resource update. However, by this logic, it may be that a user reading the value of a resource should not trigger handle_internal_event() at all.